core/types/bor_receipt.go

```diff
@@ -1,109 +1,73 @@
+	return common.BytesToHash(crypto.Keccak256(receiptKey))
 }
 
-// DecodeRLP implements rlp.Decoder, and loads both consensus and implementation
-// fields of a receipt from an RLP stream.
-func (r *BorReceiptForStorage) DecodeRLP(s *rlp.Stream) error {
-	// Retrieve the entire receipt blob as we need to try multiple decoders
-	blob, err := s.Raw()
-	if err != nil {
-		return err
-	}
-
-	return decodeStoredBorReceiptRLP(r, blob)
+// NewBorTransaction create new bor transaction for bor receipt
+func NewBorTransaction() *Transaction {
+	return NewTransaction(0, common.Address{}, big.NewInt(0), 0, big.NewInt(0), make([]byte, 0))
 }
 
-func decodeStoredBorReceiptRLP(r *BorReceiptForStorage, blob []byte) error {
-	var stored storedBorReceiptRLP
-	if err := rlp.DecodeBytes(blob, &stored); err != nil {
-		return err
+// DeriveFieldsForBorReceipt fills the receipts with their computed fields based on consensus
+// data and contextual infos like containing block and transactions.
+func DeriveFieldsForBorReceipt(receipt *Receipt, hash common.Hash, number uint64, receipts Receipts) error {
+	// get derived tx hash
+	txHash := GetDerivedBorTxHash(BorReceiptKey(number, hash))
+	txIndex := uint(len(receipts))
+
+	// set tx hash and tx index
+	receipt.TxHash = txHash
+	receipt.TransactionIndex = txIndex
+	receipt.BlockHash = hash
+	receipt.BlockNumber = big.NewInt(0).SetUint64(number)
+
+	logIndex := 0
+	for i := 0; i < len(receipts); i++ {
+		logIndex += len(receipts[i].Logs)
 	}
 
-	r.Logs = make([]*Log, len(stored.Logs))
-	for i, log := range stored.Logs {
-		r.Logs[i] = (*Log)(log)
+	// The derived log fields can simply be set from the block and transaction
+	for j := 0; j < len(receipt.Logs); j++ {
+		receipt.Logs[j].BlockNumber = number
+		receipt.Logs[j].BlockHash = hash
+		receipt.Logs[j].TxHash = txHash
+		receipt.Logs[j].TxIndex = txIndex
+		receipt.Logs[j].Index = uint(logIndex)
+		logIndex++
 	}
-	r.Bloom = BytesToBloom(LogsBloom(r.Logs).Bytes())
-
 	return nil
 }
 
-// DeriveFields fills the receipts with their computed fields based on consensus
-// data and contextual infos like containing block and transactions.
-func (r *BorReceipt) DeriveFields(hash common.Hash, number uint64) error {
-	// txHash := common.BytesToHash(crypto.Keccak256(append([]byte("bor-receipt-"), hash.Bytes()...)))
-
-	// The derived log fields can simply be set from the block and transaction
-	for j := 0; j < len(r.Logs); j++ {
-		r.Logs[j].BlockNumber = number
-		r.Logs[j].BlockHash = hash
-		// r.Logs[j].TxHash = txHash
-		r.Logs[j].TxIndex = uint(0)
-		r.Logs[j].Index = uint(j)
-	}
-	return nil
+// MergeBorLogs merges receipt logs and block receipt logs
+func MergeBorLogs(logs []*Log, borLogs []*Log) []*Log {
+	result := append(logs, borLogs...)
+	sort.SliceStable(result, func(i int, j int) bool {
+		return (result[i].BlockNumber*100000 + uint64(result[i].Index)) < (result[j].BlockNumber*100000 + uint64(result[j].Index))
```

Comment:
Why did we multiply `blocknumber` with 10^5 ? I think we can make it compile time constant.

Reply:
We are sorting by `block * 100000 + logIndex`. A simple hack to sort easily. I will make it constant.

Reply:
We need to  make `10 ** 5` constant.